beacon-chain/rpc/eth/validator/handlers.go

```diff
@@ -451,3 +452,78 @@ func (s *Server) GetAttestationData(w http.ResponseWriter, r *http.Request) {
 	}
 	http2.WriteJson(w, response)
 }
+
+// ProduceSyncCommitteeContribution requests that the beacon node produce a sync committee contribution.
+func (s *Server) ProduceSyncCommitteeContribution(w http.ResponseWriter, r *http.Request) {
+	ctx, span := trace.StartSpan(r.Context(), "validator.ProduceSyncCommitteeContribution")
+	defer span.End()
+
+	subIndex := r.URL.Query().Get("subcommittee_index")
+	index, valid := shared.ValidateUint(w, "Subcommittee Index", subIndex)
+	if !valid {
+		return
+	}
+	rawSlot := r.URL.Query().Get("slot")
+	slot, valid := shared.ValidateUint(w, "Slot", rawSlot)
+	if !valid {
+		return
+	}
+	blockRoot := r.URL.Query().Get("beacon_block_root")
+	valid = shared.ValidateHex(w, "Beacon Block Root", blockRoot)
+	if !valid {
+		return
+	}
+	contribution, ok := s.produceSyncCommitteeContribution(ctx, w, primitives.Slot(slot), index, []byte(blockRoot))
+	if !ok {
+		return
+	}
+	response := &ProduceSyncCommitteeContributionResponse{
+		Data: &shared.SyncCommitteeContribution{
+			Slot:              strconv.FormatUint(uint64(contribution.Slot), 10),
+			BeaconBlockRoot:   hexutil.Encode(contribution.BlockRoot),
+			SubcommitteeIndex: strconv.FormatUint(contribution.SubcommitteeIndex, 10),
+			AggregationBits:   hexutil.Encode(contribution.AggregationBits),
+			Signature:         hexutil.Encode(contribution.Signature),
+		},
+	}
+	http2.WriteJson(w, response)
+}
+
+// ProduceSyncCommitteeContribution requests that the beacon node produce a sync committee contribution.
+func (s *Server) produceSyncCommitteeContribution(
+	ctx context.Context,
+	w http.ResponseWriter,
+	slot primitives.Slot,
+	index uint64,
+	blockRoot []byte,
+) (*ethpbalpha.SyncCommitteeContribution, bool) {
+	msgs, err := s.SyncCommitteePool.SyncCommitteeMessages(slot)
+	if err != nil {
+		http2.HandleError(w, "Could not get sync subcommittee messages: "+err.Error(), http.StatusInternalServerError)
+		return nil, false
+	}
+	if msgs == nil {
```

Comment:
if len(msgs) == 0 is safer because that would check against an empty list that is not nil